game.cpp

```diff
@@ -1,2 +1,84 @@
 #include <bits/stdc++.h>
-using namespace std;
\ No newline at end of file
+#include "main-character.cpp"
+#include "monster-generator.cpp"
+#include "battle.cpp"
+#include "store.cpp"
+#include "ranking.cpp"
+
+using namespace std;
+
+const string INTRODUCTION = "BEM VINDO AO XIAQ\n";
+const string CONGRATULATIONS_MESSAGE = "Parabéns!! Você ganhou mais uma batalha..\n";
+const string START_OPTIONS_MESSAGE = "Digite 1 caso queira batalhar, 2 se quiser ver as instruções ou 3 para visualizar o ranking.\n";
+const string IN_GAME_OPTIONS_MESSAGE = "Digite 2 caso deseje ir a loja ou 1 se quiser continuar batalhando.\n";
+const string LOSS_MESSAGE = "Infelizmente chegamos ao fim dessa partida após essa derrota.\n";
+const string INSTRUCTIONS = "\n";
+const int SCORE_INCREMENT = 5;
+
+void play(main_character &character) {
+  //Select the difficult level before start the game.
+  int battles_counter = 0;
+  int still_in_battle = 0;
+  int current_score = 0;
+  do {
+    if(still_in_battle) {
+      printf("%s", CONGRATULATIONS_MESSAGE.c_str());
+      character.coins ++;
+      current_score += SCORE_INCREMENT;
+    }
+
+    if(battles_counter == 5) {
+      printf("%s", IN_GAME_OPTIONS_MESSAGE.c_str());
+
+      int option;
+      scanf("%d", &option);
+      if(option == 2) {
+        see_store(character);
+      }
+      battles_counter = 0;
+    }
+
+    monster enemy = generate_monster(false, ++character.game_progress_multiplier);
```

Comment:
if battles_counter % 5 == 4, set the boolean in generate_monster as true, to generate a boss

Reply:
We're not going to implement this feature anymore, it isn't in the specification

Reply:
Still, its useful to have the parameter passed correctly, if its being passed. Otherwise, we should remove the parameter at the method signature.

Reply:
I know, but I think it's not in this pr's scope. I made this to make it work, later, we improve this approach...